src/lib/components/forms/form/form-components/datepicker/datepicker.component.ts

```diff
@@ -0,0 +1,97 @@
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, Inject, Input } from '@angular/core';
+import { noop } from 'rxjs';
+import { TerraPlacementEnum } from 'src/lib/helpers/enums/terra-placement.enum';
+import { DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE, MatDateFormats } from '@angular/material/core';
+import {
+    MAT_MOMENT_DATE_ADAPTER_OPTIONS,
+    MAT_MOMENT_DATE_FORMATS,
+    MomentDateAdapter
+} from '@angular/material-moment-adapter';
+import { DatepickerInterface } from './datepicker.interface';
+import { L10N_LOCALE, L10nLocale } from 'angular-l10n';
+import * as moment from 'moment';
+import { isMoment, Moment } from 'moment';
+
+@Component({
+    selector: 'tc-datepicker',
+    templateUrl: './datepicker.component.html',
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: DatePickerComponent,
+            multi: true
+        },
+        {
+            provide: DateAdapter,
+            useClass: MomentDateAdapter,
+            deps: [MAT_DATE_LOCALE, MAT_MOMENT_DATE_ADAPTER_OPTIONS]
+        },
+        {
+            provide: MAT_DATE_LOCALE,
+            useFactory: (locale: L10nLocale) => locale.language,
```

Comment:
Ich sehe gerade, dass diese Zeile hier zu nem Buildfehler führt. 
<img width="762" alt="Bildschirmfoto 2021-04-23 um 16 30 05" src="https://user-images.githubusercontent.com/26869118/115886417-32694c80-a451-11eb-8f0b-b4c5a1e78bd2.png">

Um den zu fixen, musst du die factory-function extrahieren
```
export function matDateLocaleFactory(locale: L10nLocale) {
     return locale.language;
}
```
Und dann hier referenzieren
```suggestion
            useFactory: matDateLocaleFactory,
```